apps/passport-server/src/services/telegramService.ts

```diff
@@ -409,43 +415,136 @@ export class TelegramService {
         const messageThreadId = ctx.update.message?.message_thread_id;
         span?.setAttributes({ topicName, messageThreadId, chatId });
 
-        if (!messageThreadId)
-          return logger(
-            `[TELEGRAM] ignoring edit for general topic ${topicName}`
-          );
-
         if (!chatId || !topicName)
           throw new Error(`Missing chatId or topic name`);
 
-        const topicsForChat = await fetchTelegramTopicsByChatId(
+        const topic = await fetchTelegramTopic(
           this.context.dbPool,
-          chatId
-        );
-
-        const topic = topicsForChat.find(
-          (e) => e.topic_id?.toString() === messageThreadId?.toString()
+          chatId,
+          messageThreadId || 0
         );
 
         if (!topic) {
-          logger(`[TELEGRAM] editing topic and adding to db`);
+          logger(`[TELEGRAM] adding topic ${topicName} to db`);
+          await insertTelegramChat(this.context.dbPool, chatId);
           await insertTelegramTopic(
             this.context.dbPool,
             chatId,
-            messageThreadId,
+            messageThreadId || 0,
             topicName,
             false
           );
         } else {
-          logger(`[TELEGRAM] editing topic and updating db`);
+          logger(`[TELEGRAM] updating topic ${topicName} in db`);
           await insertTelegramTopic(
             this.context.dbPool,
             chatId,
-            messageThreadId,
+            messageThreadId || 0,
             topicName,
             topic.is_anon_topic
           );
         }
-        logger(`[TELEGRAM] Updated topic ${topicName} in the db`);
+      });
+    });
+
+    this.authBot.command("receive", async (ctx) => {
+      if (isDirectMessage(ctx))
+        return ctx.reply(`/receive can only be run in a group chat`);
+
+      const messageThreadId = ctx.message?.message_thread_id;
+
+      if (!ctx.from?.username)
+        return ctx.reply(`No username found`, {
+          reply_to_message_id: messageThreadId
+        });
+
+      if (!ALLOWED_TICKET_MANAGERS.includes(ctx.from.username))
+        return ctx.reply(
+          `Only Zupass team members are allowed to run this command.`,
+          { reply_to_message_id: messageThreadId }
+        );
+
+      // Look up topic.
+      const topic = await fetchTelegramTopic(
+        this.context.dbPool,
+        ctx.chat.id,
+        messageThreadId || 0
+      );
+      const chatId = ctx.chat.id;
+      const isReceving = true;
+
+      // If the topic doesn't exist, add it and the chatId to the DB.
+      if (!topic) {
+        logger(`[TELEGRAM] topic not found to mark as receiving.`);
+        const topicName =
+          ctx.message?.reply_to_message?.forum_topic_created?.name;
+
+        await insertTelegramChat(this.context.dbPool, chatId);
+        await insertTelegramTopic(
+          this.context.dbPool,
+          chatId,
+          messageThreadId || 0,
+          topicName || "NOT FOUND",
```

Comment:
can we ever have a topic without a name? maybe if it's the general topic, and in that case i'm not sure if "NOT FOUND" is the right thing to insert compared to just NULL

Reply:
Just throw an error if no "topic name"